src/main/java/uk/gov/hmcts/ccd/auditlog/AuditRepository.java

```diff
@@ -0,0 +1,15 @@
+package uk.gov.hmcts.ccd.auditlog;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuditRepository {
```

Comment:
better to have an AuditRepository interface with at the moment one implementation LoggerAuditRepository. 
This will allow in the future to easily change the destination of the log files if needed. For example they might want to ask to store on DB rather than on std out

Reply:
at any point of time we've one implementation only right? Also this is not a library class. https://martinfowler.com/bliki/InterfaceImplementationPair.html

Reply:
The benefit of introducing an interface is not so that we can support more than one destination at a single time. That is not what I meant. The benefit is that it allows us to change the destination limiting the amount of changes we need to do. We just inject a different implementation class without having to change the client classes using the AuditRepository. Because the client classes use an interface. If you look at any implementation of the repository design pattern on internet it's always defined as an interface, also when there is only 1 implementation class:
https://thoughts-on-java.org/implementing-the-repository-pattern-with-jpa-and-hibernate/
https://medium.com/@krzychukosobudzki/repository-design-pattern-bc490b256006

Reply:
haha.. we can't generalise every single principle to each scenario. This particular one has just one line of code (log statement), surely there won't be any caching required, dependency injection still possible.  I've seen many well known libraries including Spring has single implementation class without interface.  

Personally I disagree in this particular scenario but pushed as requested.